crates/primitives/src/signature/sig.rs

```diff
@@ -64,43 +44,52 @@ impl FromStr for Signature<()> {
     }
 }
 
-impl From<&crate::Signature> for [u8; 65] {
+impl From<&Signature> for [u8; 65] {
     #[inline]
-    fn from(value: &crate::Signature) -> [u8; 65] {
+    fn from(value: &Signature) -> [u8; 65] {
         value.as_bytes()
     }
 }
 
-impl From<crate::Signature> for [u8; 65] {
+impl From<Signature> for [u8; 65] {
     #[inline]
-    fn from(value: crate::Signature) -> [u8; 65] {
+    fn from(value: Signature) -> [u8; 65] {
         value.as_bytes()
     }
 }
 
-impl From<&crate::Signature> for Vec<u8> {
+impl From<&Signature> for Vec<u8> {
     #[inline]
-    fn from(value: &crate::Signature) -> Self {
+    fn from(value: &Signature) -> Self {
         value.as_bytes().to_vec()
     }
 }
 
-impl From<crate::Signature> for Vec<u8> {
+impl From<Signature> for Vec<u8> {
     #[inline]
-    fn from(value: crate::Signature) -> Self {
+    fn from(value: Signature) -> Self {
         value.as_bytes().to_vec()
     }
 }
 
 #[cfg(feature = "k256")]
-impl From<(k256::ecdsa::Signature, k256::ecdsa::RecoveryId)> for Signature<k256::ecdsa::Signature> {
+impl From<(k256::ecdsa::Signature, k256::ecdsa::RecoveryId)> for Signature {
     fn from(value: (k256::ecdsa::Signature, k256::ecdsa::RecoveryId)) -> Self {
         Self::from_signature_and_parity(value.0, value.1).unwrap()
     }
 }
 
+#[cfg(feature = "k256")]
+impl TryFrom<Signature> for k256::ecdsa::Signature {
+    type Error = k256::ecdsa::Error;
+
+    fn try_from(value: Signature) -> Result<Self, Self::Error> {
+        Self::from_scalars(value.r.to_be_bytes(), value.s.to_be_bytes())
```

Comment:
recovery becomes more expensive because this is not a free conversion

Reply:
worth looking into easier ways to construct the scalars without going through big endian bytes

Reply:
I don't think k256 Signature allows instantiation without going through bytes

Given that signature fields are not public, should we consider adding this to it:
```rust
#[cfg(feature = "k256")]
signature: Option<k256::ecdsa::Signature>,
```

which will be set to `None` if invalid signature is instantiated, and recovery will just fail if its `None`

Reply:
> I don't think k256 Signature allows instantiation without going through bytes

it doesn't

I think we should be okay with this overhead. this only affects the alloy sig -> k256 signature conversion and most common is decoding this from rlp

and getting rid of the additional footprint is more impactful, at least from reth pov